map/bookmark_manager.cpp

```diff
@@ -2698,16 +2698,12 @@ void BookmarkManager::PrepareFileForSharing(kml::GroupIdCollection && categories
     return;
   }
 
-  if (m_testModeEnabled)
-  {
-    handler(GetFileForSharing(std::move(collection)));
-  }
-  else
-  {
-    GetPlatform().RunTask(Platform::Thread::File,
-                          [collection = std::move(collection), handler = std::move(handler)]() mutable
-                          { handler(GetFileForSharing(std::move(collection))); });
-  }
+  GetPlatform().RunTask(Platform::Thread::File,
+                        [collection = std::move(collection), handler = std::move(handler), this]() mutable
+                        {
+                          handler(GetFileForSharing(std::move(collection)));
+                          m_testLatch.release();
```

Comment:
1. Production code should not have any test-related semaphores.
2. Can it be moved into the test's `handler` code which is likely passed from the test?
3. Why latch instead of semaphore?

Reply:
@biodranik I added a requested changes but stuck a bit with debug - test is executed but handler is not called so the whole test stops. I tried to debug and see that handler is submitted to UI thread queue but is not started. When it is done with test mode flag implementation the handler is submitted to the same place, but is started. And I can't find the difference - could you give a couple of hints please?